test/tests/functional/pbs_hook_management.py

```diff
@@ -0,0 +1,241 @@
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+import os
+import time
+import socket
+import textwrap
+import datetime
+from tests.functional import *
+from tests.functional import MGR_CMD_SET
+from tests.functional import SERVER
+from tests.functional import ATTR_h
+from tests.functional import TEST_USER
+from tests.functional import Job
+from tests.functional import JOB
+
+
+@tags('hooks', 'smoke')
+class TestHookManagement(TestFunctional):
+
+    def test_hook_00(self):
+        """
+        By creating an import hook, it executes a management hook.
+        """
+        self.logger.info("**************** HOOK START ****************")
+        # hook_name = "test_management"
+        hook_name = "management"
+        hook_msg = 'running management hook_00'
+        hook_body = """
+        import pbs
+        e = pbs.event()
+        m = e.management
+        pbs.logmsg(pbs.LOG_DEBUG, '%s')
+        """ % hook_msg
+        hook_body = textwrap.dedent(hook_body)
+        # attrs = {'event': 'management', 'enabled': 'True'}
+        # rv = self.server.create_import_hook(
+        #     hook_name, attrs=attrs, body=hook_body, overwrite=True)
+
+        qmgr_path = os.path.join(self.server.pbs_conf["PBS_EXEC"], "bin",
+                                 "qmgr")
+        fn00 = self.du.create_temp_file()
+        fn01 = self.du.create_temp_file()
+        with open(fn00, "w+") as tempfd:
+            tempfd_name = tempfd.name
+            tempfd.write(hook_body)
+            self.logger.info("tempfd_name:%s" % tempfd_name)
+
+        self.logger.info("hook_name:%s" % hook_name)
+        qmgr_setup = [qmgr_path, "-c",
+                      "create hook %s event=management" % hook_name]
+        qmgr_cmd = [qmgr_path, "-c",
+                    "import hook %s application/x-python default %s" %
+                    (hook_name, tempfd_name)]
+        qmgr_cleanup = [qmgr_path, "-c",
```

Comment:
You don't have to worry about cleanup, setup() will take care of it, next time another test starts.

Reply:
Since the qmgr commands are firing hooks now, I wanted to test the delete also.